client/src/app/store/effects/app.effects.ts

```diff
@@ -80,10 +100,24 @@ export class Effects {
         return this.userService.userLogin(action.user).pipe(
           map((data) => appActions.setCurrentUser({ user: data })),
           catchError((data) =>
-            of(appActions.setAppError({ errorMessage: data }))
+            of(
+              appActions.setAppError({
+                errorMessage: mirrorError(data),
+              })
+            )
           )
         );
       })
     )
   );
 }
+
+function mirrorError(backEndError): AppError {
+  const errorMessage: AppError = {
+    status: backEndError.error.status,
+    timestamp: backEndError.error.timestamp,
+    debugMessage: backEndError.error.debugMessage,
+    message: backEndError.error.message,
```

Comment:
As a defensive programming technique, we always need to assume that there is a possibility a chain of objects doesn't exist.

if (backendError && backendError.error) then ... else return null

Also, this is minor but can you change 'backEndError' to 'backendError'